File: Classes/EVEOnlineAPI/Character/EVEMailingLists.m
//
//  EVEMailingLists.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 6/4/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEMailingLists.h"

@implementation EVEMailingListsItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"listID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"displayName":@{@"type":@(EVEXMLSchemePropertyTypeString)}};
	return scheme;
}

@end


@implementation EVEMailingLists
@synthesize mailingListsMap = _mailingListsMap;

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"mailingLists":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMailingListsItem class]}};
	
	return scheme;
}

- (NSDictionary*) mailingListsMap {
	if (!_mailingListsMap) {
		_mailingListsMap = [[NSMutableDictionary alloc] initWithCapacity:self.mailingLists.count];
		for (EVEMailingListsItem* item in self.mailingLists)
			((NSMutableDictionary*)_mailingListsMap)[@(item.listID)] = item;
	}
	return _mailingListsMap;
}

@end

File: EVEAPI/EVEAPI/EVEMailingLists.swift
//
//  EVEMailingLists.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEMailingListsItem: EVEObject {
	public var listID: Int = 0
	public var displayName: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"listID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
			"displayName":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}


public class EVEMailingLists: EVEResult {
	public var mailingLists: [EVEMailingListsItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"mailingLists":EVESchemeElementType.Rowset(elementName: nil, type: EVEMailingListsItem.self, transformer: nil),
		]
	}
}

